### PR TITLE
Wel details

### PR DESCRIPTION
**Pull request by djaglowski**
## Description of Changes
- Addresses #147 
- Includes unit testing based on specific examples

### Limitations
- The parsing method is quite complicated, and based heavily on state that is difficult to reason about. I don't love the implementation, but I'm confident that it handles the known set of examples well enough.
- Messages are not parsed to arbitrary depth. If this ends up being necessary, it'll require some refactoring.
- At the maximum depth that this implementation handles, there is a lot of inconsistency in message structure. As a result, these values are not parsed, but just treated as strings.

## **Please check that the PR fulfills these requirements**
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)
- [x] Add a changelog entry (for non-trivial bug fixes / features)
- [x] CI passes

### CODE REVIEW

**Comment by djaglowski** [comment]
| Log Files | Logs / Second | CPU Avg (%) | CPU Avg Δ (%) | Memory Avg (MB) | Memory Avg Δ (MB) |
| --- | --- | --- | --- | --- | --- |
|1|1000|1.6724602|+0.051736116|123.73707|-4.3495407|
|1|5000|6.5863624|+0.48280764|135.22765|-1.478302|
|1|10000|12.034732|-0.8104105|143.59174|-1.8941193|
|1|50000|57.86202|-3.466652|172.66716|-9.809265|
|1|100000|114.70694|-5.591507|220.6716|-21.962158|
|10|100|2.3449028|+0.06901336|132.33163|-0.45257568|
|10|500|6.8966856|-0.3104558|136.6933|-1.4854431|
|10|1000|13.172718|-0.65516186|146.25485|+0.38241577|
|10|5000|67.77386|+0.92813873|182.6584|-1.1920776|
|10|10000|127.38031|+0.23866272|224.58823|-1.5511932|

**Comment by djaglowski** [comment]
| Log Files | Logs / Second | CPU Avg (%) | CPU Avg Δ (%) | Memory Avg (MB) | Memory Avg Δ (MB) |
| --- | --- | --- | --- | --- | --- |
|1|1000|3.2586505|+1.6379265|127.25458|-0.83203125|
|1|5000|6.379513|+0.27595806|130.90369|-5.8022614|
|1|10000|12.0691185|-0.77602386|143.82921|-1.6566467|
|1|50000|59.881096|-1.4475746|180.4363|-2.0401306|
|1|100000|113.7608|-6.5376434|226.22293|-16.410828|
|10|100|2.5517864|+0.27589703|131.5004|-1.2838135|
|10|500|7.4485908|+0.24144936|137.95056|-0.22817993|
|10|1000|13.483042|-0.34483814|146.30698|+0.4345398|
|10|5000|65.519585|-1.3261337|183.69855|-0.15193176|
|10|10000|126.10399|-1.0376587|223.01886|-3.1205597|

File: operator/builtin/input/windows/security.go [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+				details[l.k] = list
+			}
+		}
+	}
+
+	if len(moreInfo) > 0 {
+		details["Additional Context"] = moreInfo
+	}
+
+	if len(unparsed) > 0 {
```
Comment (camdencheek) [inline operator/builtin/input/windows/security.go:102]: Looks like `unparsed` is never modified from its initial empty slice

File: operator/builtin/input/windows/security.go [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+	mp.ptr++
+}
+
+func (mp *messageProcessor) hasNext() bool {
+	return mp.ptr < len(mp.lines)
+}
+
+func countIndent(line string) int {
+	i := 1
+	for pre := strings.Repeat("\t", i); strings.HasPrefix(line, pre); pre = strings.Repeat("\t", i) {
```
Comment (camdencheek) [inline operator/builtin/input/windows/security.go:198]: This might be one of the more creative for loop constructs I've seen

**Review by camdencheek — APPROVED** [review]

**Comment by djaglowski** [comment]
| Log Files | Logs / Second | CPU Avg (%) | CPU Avg Δ (%) | Memory Avg (MB) | Memory Avg Δ (MB) |
| --- | --- | --- | --- | --- | --- |
|1|1000|1.672419|+0.05169487|124.748116|-3.3384933|
|1|5000|5.931152|-0.17240286|136.80603|+0.1000824|
|1|10000|12.103714|-0.7414284|141.68535|-3.8005066|
|1|50000|58.208572|-3.120098|174.7438|-7.7326202|
|1|100000|115.07038|-5.2280655|229.28017|-13.353592|
|10|100|2.4139705|+0.13808107|131.96121|-0.82299805|
|10|500|6.689757|-0.5173845|138.12581|-0.05293274|
|10|1000|13.552072|-0.27580833|143.75809|-2.1143494|
|10|5000|62.275448|-4.5702705|179.65639|-4.194092|
|10|10000|140.84818|+13.706528|237.00162|+10.862198|

**Pull request merged**